libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp

```diff
@@ -165,21 +165,24 @@ const AP_Param::GroupInfo AP_AdvancedFailsafe::var_info[] = {
 // check for Failsafe conditions. This is called at 10Hz by the main
 // ArduPlane code
 void
-AP_AdvancedFailsafe::check(bool geofence_breached, uint32_t last_valid_rc_ms)
-{    
+AP_AdvancedFailsafe::check(uint32_t last_valid_rc_ms)
+{
     if (!_enable) {
         return;
     }
 
+#if AP_FENCE_ENABLED
     // we always check for fence breach
     if(_enable_geofence_fs) {
-        if (geofence_breached || check_altlimit()) {
+        const AC_Fence *ap_fence = AP::fence();
+        if ((ap_fence && ap_fence->get_breaches() != 0) || check_altlimit()) {
```

Comment:
I think the style we prefer is ap_fence != nullptr

Reply:
Yeah, I'd noticed that and was just hoping to slide it through ;-)

Changed it.